evalml/automl/automl_search.py

```diff
@@ -720,15 +720,15 @@ def _get_baseline_pipeline(self):
         if self.problem_type == ProblemTypes.BINARY:
             baseline = BinaryClassificationPipeline(component_graph=["Baseline Classifier"],
                                                     custom_name="Mode Baseline Binary Classification Pipeline",
-                                                    custom_hyperparameters={"strategy": ["mode"]})
```

Comment:
This was the odd dask issue that came up--it wasn't happy with this line. This is incorrect anyways (doesnt specify which component) but funny that this didn't come up in other tests... I think it's because the default is mode/mean anyways so automl just did its thing and ignored this, but now that i'm adding custom hyperparameters to repr, it bugged out 😂 

I think it makes sense to change this

Reply:
Do you have the dask stacktrace?

Reply:
Here ya go!
```

    def test_automl_max_iterations(self):
        """ Making sure that the max_iterations parameter limits the number of pipelines run. """
        X, y = self.X_y_binary
        max_iterations = 4
        par_automl = AutoMLSearch(X_train=X, y_train=y, problem_type="binary", engine=self.parallel_engine,
                                  max_iterations=max_iterations)
>       par_automl.search()

evalml\tests\automl_tests\test_automl_dask.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
evalml\automl\automl_search.py:580: in search
    self._add_baseline_pipelines()
evalml\automl\automl_search.py:753: in _add_baseline_pipelines
    computation = self._engine.submit_evaluation_job(self.automl_config, baseline, self.X_train, self.y_train)
evalml\automl\engine\dask_engine.py:89: in submit_evaluation_job
    logger=logger)
c:\users\runneradmin\miniconda3\envs\curr_py\lib\site-packages\distributed\client.py:1565: in submit
    key = funcname(func) + "-" + tokenize(func, kwargs, *args)
c:\users\runneradmin\miniconda3\envs\curr_py\lib\site-packages\dask\base.py:795: in tokenize
    return md5(str(tuple(map(normalize_token, args))).encode()).hexdigest()
c:\users\runneradmin\miniconda3\envs\curr_py\lib\site-packages\dask\utils.py:512: in __call__
    return meth(arg, *args, **kwargs)
c:\users\runneradmin\miniconda3\envs\curr_py\lib\site-packages\dask\base.py:806: in normalize_dict
    return normalize_token(sorted(d.items(), key=str))
evalml\pipelines\pipeline_base.py:528: in __repr__
    custom_hyperparameters_repr = ', '.join([f"'{component}':{{{repr_component(hyperparameters)}}}," for component, hyperparameters in self.custom_hyperparameters.items()]) if self.custom_hyperparameters else None
evalml\pipelines\pipeline_base.py:528: in <listcomp>
    custom_hyperparameters_repr = ', '.join([f"'{component}':{{{repr_component(hyperparameters)}}}," for component, hyperparameters in self.custom_hyperparameters.items()]) if self.custom_hyperparameters else None
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

parameters = ['mode']

    def repr_component(parameters):
>       return ', '.join([f"'{key}': {safe_repr(value)}" for key, value in parameters.items()])
E       AttributeError: 'list' object has no attribute 'items'

```

Reply:
Thanks! I wanted to know how dask ended up calling repr